subsys/nrf_rpc/nrf_rpc_uart.c

```diff
@@ -171,26 +187,35 @@ static int init(const struct nrf_rpc_tr *transport, nrf_rpc_tr_receive_handler_t
 	return 0;
 }
 
+static void send_byte(const struct device *dev, uint8_t byte)
+{
+
+		if (byte == hdlc_char_delimiter || byte == hdlc_char_escape) {
+			uart_poll_out(dev, hdlc_char_escape);
+			byte ^= 0x20;
+		}
+
+		uart_poll_out(dev, byte);
+}
+
 static int send(const struct nrf_rpc_tr *transport, const uint8_t *data, size_t length)
 {
+	uint16_t crc = crc16_ccitt(0xffff, data, length);
 	LOG_HEXDUMP_DBG(data, length, "Sending frame");
+	LOG_HEXDUMP_DBG((uint8_t*)&crc, CRC_SIZE, "frame crc");
 	struct nrf_rpc_uart *uart_tr = transport->ctx;
 
 	k_sem_take(&uart_tr->uart_tx_sem, K_FOREVER);
 
 	uart_poll_out(uart_tr->uart, hdlc_char_delimiter);
 
 	for (size_t i = 0; i < length; i++) {
-		uint8_t byte = data[i];
-
-		if (data[i] == hdlc_char_delimiter || data[i] == hdlc_char_escape) {
-			uart_poll_out(uart_tr->uart, hdlc_char_escape);
-			byte ^= 0x20;
-		}
-
-		uart_poll_out(uart_tr->uart, byte);
+		send_byte(uart_tr->uart, data[i]);
 	}
 
```

Comment:
```
crc = sys_cpu_to_le16(crc);
```

Reply:
I think I don't understand this comment. `data[i]` is actually a single byte. Have you meant it for 222 line?

Reply:
Some lines are shifted, but you applied fix. 👍🏻

Reply:
This sems wrong now. `CRC` is `uint16_t` value that independent of the endiannness, and then it is put in the frame in little endian, so `sys_cpu_to_le16` isn't necessary.

Reply:
Whether the value should be converted to the protocol endianness before serialization?

Reply:
This would be true if we sent `uint16_t` cast to `uint8_t*`. In this case, we explicitly send the least significant byte first: `crc & 0xff`, thus no bswap is needed before.

Reply:
Seems, you're right.

Reply:
@alxelax @Damian-Nordic I have changed the proposed functions to [this](https://docs.zephyrproject.org/latest/doxygen/html/sys_2byteorder_8h.html#a68df8f14dfc0d8715d16ac87dd4c73d2) and [this](https://docs.zephyrproject.org/latest/doxygen/html/sys_2byteorder_8h.html#a91fae25b12126c2b35d240f56866184d). According to documentation those are endianess safe (storing and getting data in Big Endian or network byte order) and are alignment safe as well.

Reply:
Any reason Big Endian instead of Little Endian is used? I can see OpenThread and PPP driver uses LE for HDLC(-lite) frames so maybe would be more intuitive to align?

Reply:
When comes to networking big endian is more intuitive.

Reply:
Yes, but when it comes to HDLC, this is what RFC includes:
> Frame Check Sequence (FCS) Field

> The Frame Check Sequence field defaults to 16 bits (two octets). The FCS is transmitted least significant octet first, which contains the coefficient of the highest term.

Reply:
fixed.

Reply:
But feel free to disregard this comment if you think we should diverge from HDLC at this.